tools/install-powershell.ps1

```diff
@@ -137,12 +137,16 @@ try {
 
         # Setting Tls to 12 to prevent the Invoke-WebRequest : The request was
         # aborted: Could not create SSL/TLS secure channel. error.
-        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+        $originalValue = [Net.ServicePointManager]::SecurityProtocol
+        [Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::Tls12
```

Comment:
This should be tls1.0, tls1.1 and tls1.2.  Then no need to revert

Reply:
I'm torn. Enabling it in the boot strap function makes sense because we are setting up the PowerShell environment for builds and tests. Leaving the change in the install code means we are modifying the user's PowerShell environment. That's not exactly what I would expect (even though it is a non-breaking environment change that arguably increases their security....).

Reply:
It is what the defaults should be... you could argue that 1.0 should not be included.  I included it to be more compatible.  If you are connecting to anything older, there are serious issues.

Reply:
I agree that's what the default settings _should_ be.... unfortunately, they are not the defaults. It is possible that changing these settings without reverting (and in a way that is somewhat hidden from the user) could potentially break their environment. There are some sites that di respond differently on TLS 1.0, 1.1, and 1.2. That's why I think the context matters as to whether the settings changes are left or reverted. `Start-PSBootStrap` is a "configure my environment to be able to build, debug, and test PowerShell" function where as `install-powershell.ps1` is a script for adding pwsh to the environment. In the former I expect the environment to change, in the later, the only environment change I expect is to have pwsh installed. *shrug